File: 프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.java
class Solution {
    public String solution(String[] my_strings, int[][] parts) {
        StringBuilder sb = new StringBuilder();
        
        for(int i=0; i<parts.length; i++) {
            int[] range = parts[i];
            int start = range[0];
            int end = range[1];
            
            sb.append(my_strings[i].substring(start, end + 1));
        }
        
        return sb.toString();
    }
}

File: 프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.js
function solution(my_strings, parts) {
    // let answer = '';
    // for(let i=0; i<my_strings.length; i++) {
    //     let [s, e] = parts[i];
    //     answer += my_strings[i].slice(s, e+1);
    // }
    // return answer;
    
    // return parts.map(([s, e], i) => {
    //     return my_strings[i].slice(s, e+1);
    // }).join('');
    
    return my_strings.map((str, i) => {
        let [s, e] = parts[i];
        return str.slice(s, e+1);
    }).join('');
}